karapace/protobuf/schema.py

```diff
@@ -106,6 +109,49 @@ def option_element_string(option: OptionElement) -> str:
     return f"option {result};\n"
 
 
+@dataclasses.dataclass(frozen=True)
```

Comment:
nit (non-blocking): There exists a `@default_dataclass` that could be used here.